### PR TITLE
record level permissions: added POST endpoints

### PR DESCRIPTION
**Pull request by shriramshankar**
/v1/[subjects|aspects|perspectives|lenses]/{key}/writers endpoints

### CODE REVIEW

File: api/v1/swagger.yaml [hunk 1]
```diff
@@ -1597,6 +1643,52 @@ paths:
           $ref: "#/responses/404"
         default:
           $ref: "#/responses/genericError"
+    post:
+        security:
+          - jwt: []
+        summary: Add one or more users to an lens' list of authorized writers
```
Comment (iamigo) [inline api/v1/swagger.yaml]: Add one or more users to **a** lens' list of authorized writers

File: api/v1/swagger.yaml [hunk 2]
```diff
@@ -1597,6 +1643,52 @@ paths:
...
+        tags: [ lenses ]
+        description: >-
+          Add one or more users to an lens' list of authorized writers. If the Refocus configuration parameter `useAccessToken` is set to `true`, you must include an `Authorization` request header with your [JSON Web Token](https://tools.ietf.org/html/rfc7519) (JWT) as the value. You can get a token using `POST /v1/register` or `POST /v1/token`.
```
Comment (iamigo) [inline api/v1/swagger.yaml]: Add one or more users to **a** lens' list...

File: api/v1/swagger.yaml [hunk 3]
```diff
@@ -2133,7 +2225,52 @@ paths:
           $ref: "#/responses/404"
         default:
           $ref: "#/responses/genericError"
-
+    post:
+      security:
+        - jwt: []
+      summary: Add one or more users to an perspective's list of authorized writers
```
Comment (iamigo) [inline api/v1/swagger.yaml]: ... users to **a** ...

File: api/v1/swagger.yaml [hunk 4]
```diff
@@ -2133,7 +2225,52 @@ paths:
...
+      tags: [ perspectives ]
+      description: >-
+        Add one or more users to an perspective's list of authorized writers. If the Refocus configuration parameter `useAccessToken` is set to `true`, you must include an `Authorization` request header with your [JSON Web Token](https://tools.ietf.org/html/rfc7519) (JWT) as the value. You can get a token using `POST /v1/register` or `POST /v1/token`.
```
Comment (iamigo) [inline api/v1/swagger.yaml]: ... users to **a**...

File: api/v1/swagger.yaml [hunk 5]
```diff
@@ -4161,6 +4298,52 @@ paths:
           $ref: "#/responses/404"
         default:
           $ref: "#/responses/genericError"
+    post:
+        security:
+          - jwt: []
+        summary: Add one or more users to an subjects list of authorized writers
+        tags: [ subjects ]
+        description: >-
+          Add one or more users to an subjects list of authorized writers. If the Refocus configuration parameter `useAccessToken` is set to `true`, you must include an `Authorization` request header with your [JSON Web Token](https://tools.ietf.org/html/rfc7519) (JWT) as the value. You can get a token using `POST /v1/register` or `POST /v1/token`.
```
Comment (iamigo) [inline api/v1/swagger.yaml]: ... users to **a**...

File: api/v1/swagger.yaml [hunk 6]
```diff
@@ -6482,6 +6665,110 @@ definitions:
...
+  AspectWriterResponse:
+    type: object
+    description: >
+      Users having write permission to the aspects
+    properties:
+      aspectId:
+        type: string
+        readOnly: true
+        description: >
+          Id of a writer protected aspect
```
Comment (iamigo) [inline api/v1/swagger.yaml]: Id of a write-protected aspect

File: api/v1/swagger.yaml [hunk 7]
```diff
@@ -6482,6 +6665,110 @@ definitions:
...
+      userId:
+        type: string
+        readOnly: true
+        description: >
+          Id of an user having write permission to the aspect
```
Comment (iamigo) [inline api/v1/swagger.yaml]: Id of **a** user...

File: api/v1/swagger.yaml [hunk 8]
```diff
@@ -6482,6 +6665,110 @@ definitions:
...
+  LensWriterResponse:
+    type: object
+    description: >
+      Users having write permission to the lenses
+    properties:
+      lensId:
+        type: string
+        readOnly: true
+        description: >
+          Id of a writer protected aspect
```
Comment (iamigo) [inline api/v1/swagger.yaml]: write-protected

File: api/v1/swagger.yaml [hunk 9]
```diff
@@ -6482,6 +6665,110 @@ definitions:
...
+  PerspectiveWriterResponse:
+    type: object
+    description: >
+      Users having write permission to the perspectives
+    properties:
+      perspectiveId:
+        type: string
+        readOnly: true
+        description: >
+          Id of a writer protected perspective
```
Comment (iamigo) [inline api/v1/swagger.yaml]: write-protected

File: api/v1/swagger.yaml [hunk 10]
```diff
@@ -6482,6 +6665,110 @@ definitions:
...
+      userId:
+        type: string
+        readOnly: true
+        description: >
+          Id of an user having write permission to the perspective
```
Comment (iamigo) [inline api/v1/swagger.yaml]: Id of **a** user...

File: api/v1/swagger.yaml [hunk 11]
```diff
@@ -6482,6 +6665,110 @@ definitions:
...
+  SubjectWriterResponse:
+    type: object
+    description: >
+      Users having write permission to the subject
+    properties:
+      aspectId:
+        type: string
+        readOnly: true
+        description: >
+          Id of a writer protected subject
```
Comment (iamigo) [inline api/v1/swagger.yaml]: Id of a write-protected subject

File: api/v1/swagger.yaml [hunk 12]
```diff
@@ -6482,6 +6665,110 @@ definitions:
...
+      userId:
+        type: string
+        readOnly: true
+        description: >
+          Id of an user having write permission to the subject
```
Comment (iamigo) [inline api/v1/swagger.yaml]: Id of **a** user...

**Comment by shriramshankar** [comment]
@iamigo @annyhe PR updated

**Pull request merged**